src/matchBuilder.cpp

```diff
@@ -1,9 +1,17 @@
 #include "matchBuilder.hpp"
 
+#include <iostream>
+#include <random>
+#include <string>
+
+#include "json.hpp"
+#include "mapping.hpp"
+#include "myRandom.hpp"
```

Comment:
I added these due to a principal called "Include What You Use" (IWYU). Some of these were necessary since the header now has minimal dependencies (after I trimmed it down) so even the header didn't provide these includes anymore. Even so, it's best to include at the top of the file everything that this specific file makes use of. It's easier to see and modify, and you don't have to worry about the header (or some other header) removing a dependency that this file depends on even though it didn't say that it depended on it.

It's easy to think of your headers as tightly coupled to your implementation file, but they serve separate purposes and it's best to keep dependencies of any one file documented within that file. Hence, IWYU.

Reply:
Good program structure advice. Ill take it into account moving forward and implement your fix on this.